### PR TITLE
fix: `\label` should be included in `\begin`

### PR DESCRIPTION
**Pull request by reuixiy**

### CODE REVIEW

**Comment by reuixiy** [comment]
Wait ... I found some problems...

**Comment by reuixiy** [comment]
I don't know much about LaTeX, but `\label` should be included in `\begin` according to [the examples in this page](https://en.wikibooks.org/wiki/LaTeX/Labels_and_Cross-referencing#Formulae) and [here](https://docs.mathjax.org/en/latest/input/tex/eqnumbers.html).

However, MathJax v2 seems cannot display that well...


▴ before this PR with MathJax v2


▴ after this PR with MathJax v2

Anyway, this should be fixed in MathJax v3:


▴ after this PR with MathJax v3

https://github.com/theme-next/hexo-theme-next/pull/1314

**Review by stevenjoezhang — APPROVED** [review]

**Review by stevenjoezhang — APPROVED** [review]
